crates/evm/core/src/decode.rs

```diff
@@ -176,6 +179,34 @@ impl RevertDecoder {
             return Some(std::str::from_utf8(err).unwrap().to_string());
         }
 
+        // try from https://openchain.xyz
+        let (tx, rx) = mpsc::channel();
+        let encoded_selector = hex::encode(selector);
+
+        thread::spawn(move || {
+            let rt = tokio::runtime::Runtime::new().unwrap();
+            let result = rt.block_on(decode_function_selector(&encoded_selector));
+            tx.send(result).unwrap();
+        });
```

Comment:
this is a bit expensive to do per invocation.

I can see how this could be useful, but I'd like to make this configurable this behaviour on the `RevertDecoder` level by giving it an optional OpenChainClient and only use it if we can retrieve the a tokio handle

Reply:
done